server/src/main/java/org/opensearch/index/store/Store.java

```diff
@@ -840,18 +859,15 @@ public void cleanupAndPreserveLatestCommitPoint(
     private void cleanupFiles(
         String reason,
         Collection<String> localSnapshot,
-        @Nullable Collection<String> additionalFiles,
-        boolean deleteTempFiles
+        @Nullable Collection<String> additionalFiles
     ) throws IOException {
         assert metadataLock.isWriteLockedByCurrentThread();
         for (String existingFile : directory.listAll()) {
             if (Store.isAutogenerated(existingFile)
                 || localSnapshot != null && localSnapshot.contains(existingFile)
                 || (additionalFiles != null && additionalFiles.contains(existingFile))
                 // also ensure we are not deleting a file referenced by an active reader.
-                || replicaFileTracker != null && replicaFileTracker.canDelete(existingFile) == false
-                // prevent temporary file deletion during reader cleanup
-                || deleteTempFiles == false && existingFile.startsWith(REPLICATION_PREFIX)) {
```

Comment:
We still need to check if the file starts with the tmp prefix.  I think tmp file deletion should be done by MultiFileWriter only.  Or move the check to `canDelete` in ReplicaFileTracker.

Reply:
Added back logic to prevent temporary file deletion.